_posts/2022-04-26-fivetran-custom-connector.md

```diff
@@ -0,0 +1,335 @@
+---
+layout: post
+title: FiveTran Paginated Custom Connector
+date: 2022-05-26
+tags: fivetran custom connector custom-connector pagination getmore state management
+author: Joseph Miocevich
+image: img/fivetran_logo.png
+description:
+  Overview of a custom fivetran connector with pagination
+---
+
+![Header](/img/fivetran_logo.png)
+
+<br/>
+
+## Overview
+
+In simple terms, a FiveTran custom connector a cloud hosted function that can be setup on a cloud platform of your choice, which allows you to load data into FiveTran with a API FiveTran does not natively support. The benefits of this is that these connectors will have the same capabilities of standard connectors:
+
+- Incremental updates
+- Source data type inference
+- Automatic schema updates
+- Data de-deduplication
+- Destination ingestion optimisations
+- Logs and alerts to monitor events and troubleshoot issues
+
+Use FiveTran's Function connectors if:
+
+- Fivetran doesn't have a connector for your source
+- You are using private APIs or custom applications
+- You are using a source or API that Fivetran is unlikely to support in the near future
+- You want to sync unsupported file formats that require pre-processing
+- You have sensitive data that needs filtering or anonymizing before entering the destination
+
+*Fivetran* . Fivetran Custom Connector. (n.d.). Retrieved May 26, 2022, from https://fivetran.com/docs/functions 
+
+<br>
+
+***
+<br>
+## General Principles
+
+**State Management**
+
+State is a JSON object that contains cursors from the previous successful FiveTran executions run.
+
+[https://fivetran.com/docs/functions/faq/use-state-object](https://fivetran.com/docs/functions/faq/use-state-object) 
+
+[https://fivetran.com/docs/functions/faq/use-secrets-object](https://fivetran.com/docs/functions/faq/use-secrets-object)
+
+
+**Guardrails to consider**
+
+- Lambda timeouts and payload limit
+- Ensure calls to the Lambda are **idempotent** for a given FiveTran cursor
+- Be wary or avoid using cursors supplied by the pagination of an upstream API that may not be persistent.
+
+
+**FiveTran Pagination**
+
+FiveTran pagination allows your lambda function to specify if there is more data to be collected, this is achieved using the hasMore boolean return.
+
+When hasMore = True ,state is updated as normal, however FiveTran immediately calls the lambda with the updated state. This will keep occurring until hasMore = False, which then resets FiveTran to its default state.
```

Comment:
```suggestion
When `hasMore = True` ,state is updated as normal, however FiveTran immediately calls the lambda with the updated state. This will keep occurring until `hasMore = False`, which then resets FiveTran to its default state.
```